src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        toResize();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        toResize();
+        if (index != size) {
+            if (index >= size || index < 0) {
+                throw new ArrayListIndexOutOfBoundsException("Incorrect index: " + index);
+            }
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        }
+        elementData[index] = value;
+        size++;
```

Comment:
what if index == size ? I think it'll cause IndexOutOfBoundsException (in case your toResize() method has the second condition within 'if' block removed). Think why and try to fix it